clusterloader/cluster_loader.go

```diff
@@ -57,31 +55,51 @@ var _ = framework.KubeDescribe("Cluster Loader [Feature:ManualPerformance]", fun
 		// TODO sjug: add concurrency
 		for _, p := range project {
 			// Find tuning if we have it
-			tuning := getTuningSet(tuningSets, p.Tuning)
-			framework.Logf("Our tuning set is: %v", tuning)
+			tuning := clusterloaderframework.TuningSets(tuningSets).Get(p.Tuning)
+
+			framework.Logf("Tuning set is: %+v", tuning)
 			for j := 0; j < p.Number; j++ {
-				// Create namespaces as defined in Cluster Loader config
+				// Create namespaces as defined in the config
 				nsName := appendIntToString(p.Basename, j)
-				ns, err := f.CreateNamespace(nsName, nil)
-				framework.ExpectNoError(err)
-				framework.Logf("%d/%d : Created new namespace: %v", j+1, p.Number, nsName)
-				namespaces = append(namespaces, ns)
+				ns, err := clusterloaderframework.CreateNSIfNotExists(f, nsName)
+				if err != nil {
+					framework.Failf("Error creating NS: %v", err)
+				}
+				// Keep track of all the namespaces we have created, not too useful currently
+				namespaces = appendUnique(namespaces, ns)
 
 				// Create templates as defined
 				for _, template := range p.Templates {
-					createTemplate(template.Basename, ns, mkPath(template.File), template.Number, tuning)
+					createTemplate(template.Basename, ns, clusterloaderframework.MakePath(template.File), template.Number, tuning)
+				}
+				// RCs are a thing as well
+				for _, RC := range p.RCs {
+					config, err := RC.ParseConfig()
+					if err != nil {
+						framework.Failf("Error parsing config, %v", err)
+					}
+					label, _ := RC.ConvertToLabelSet()
+					if err = clusterloaderframework.CreateRC(f, RC.Basename, ns.Name, label, config.Spec, RC.Number); err != nil {
+						framework.Failf("Error creating RC, %v", err)
+					}
 				}
 				// This is too familiar, create pods
 				for _, pod := range p.Pods {
-					config := parsePods(mkPath(pod.File))
-					labels := map[string]string{"purpose": "test"}
-					clusterloaderframework.CreatePods(f, pod.Basename, ns.Name, labels, config.Spec, pod.Number, tuning)
+					config, err := pod.ParseConfig()
+					if err != nil {
+						framework.Failf("Error parsing config, %v", err)
+					}
+					label, _ := pod.ConvertToLabelSet()
+					clusterloaderframework.CreatePods(f, pod.Basename, ns.Name, label, config.Spec, pod.Number, tuning)
 				}
 			}
+			// Only sleeps for each new project defined in the config, need to move up to sleep for every copy
```

Comment:
OK - I don't have strong opinion about it. We can leave as is. Please split the line into two then making it a TODO: Consider ...